prime-router/src/main/kotlin/fhirengine/utils/HL7Reader.kt

```diff
@@ -118,29 +120,39 @@ class HL7Reader(private val actionLogger: ActionLogger) : Logging {
      * This function assumes all the message types will be the same if this is a HL7 batch.
      */
     private fun getMessageModelClasses(rawMessage: String): List<Class<out AbstractMessage>> {
-        val iterator = Hl7InputStreamMessageIterator(rawMessage.byteInputStream())
-        if (iterator.hasNext()) {
-            try {
-                val firstMessage = iterator.next()
-                return when (val messageType = getMessageType(firstMessage)) {
-                    "ORU" -> listOf(
-                        v27_ORU_R01::class.java,
-                        v251_ORU_R01::class.java
-                    )
-                    "OML" -> listOf(
-                        v251_OML_O21::class.java
-                    )
+        try {
+            val messageProfile = getMessageProfile(rawMessage)
+            if (messageProfile != null) {
+                when (messageProfile.typeID) {
+                    "ORU" -> {
+                        return when (messageProfile.profileID) {
+                            // TODO: NIST ELR conformance profile to be enabled in a future PR
+                            "NIST_ELR_TEST" -> listOf(
```

Comment:
To enable NIST ELR conformance profile, change this string to `"NIST_ELR"`

Reply:
Put this in the comment?

Reply:
Comments on the steps to enable the NIST ELR profile have been updated to clarify what should be changed.